applications/asset_tracker_v2/src/modules/modem_module.c

```diff
@@ -639,6 +792,26 @@ static int setup(void)
 	return 0;
 }
 
+/* Message handler for STATE_INIT */
+static void on_state_init(struct modem_msg_data *msg)
+{
+	if (IS_EVENT(msg, modem, MODEM_EVT_CARRIER_INITIALIZED)) {
+		int err;
+
+		state_set(STATE_DISCONNECTED);
+
+		err = at_cmd_init();
+		__ASSERT(err == 0, "Failed initializing at_cmd");
+
+		err = at_notif_init();
```

Comment:
Both the AT command driver and the notification manager will be initialised by the Carrier lib, I guess it's not really a problem since they will check whether they are initialised already, but at least `at_notif_init()` will throw a warning.

Reply:
I tried removing this one, but it appears that `LWM2M_CARRIER_EVENT_MODEM_INIT` is sent before the `at_cmd` and `at_notif` libraries are initialized, so the call to `setup()` a few lines down fails. So for now we'll just have to stick with the warning which actually comes when the carrier library initializes the two a bit later on.

As a future improvement, it could be considered to act on `LWM2M_CARRIER_EVENT_CONNECTING` instead of `LWM2M_CARRIER_EVENT_MODEM_INIT`, but I'll leave it as is for now.